src/lib/authentification.ts

```diff
@@ -37,65 +40,57 @@ export const CANISTER_ID_WATER_NEURON = DEV
 	: 'tsbvt-pyaaa-aaaar-qafva-cai';
 export const CANISTER_ID_ICPSWAP_POOL = 'e5a7x-pqaaa-aaaag-qkcga-cai';
 
-export interface AuthResult {
-	actors: Actors;
-	principal: Principal;
-}
+export async function connectWithInternetIdentity() {
+	try {
+		const authClient = await AuthClient.create();
 
-export interface Actors {
-	icpLedger: icpLedgerInterface;
-	nicpLedger: icrcLedgerInterface;
-	wtnLedger: icrcLedgerInterface;
-	waterNeuron: waterNeuronInterface;
-	boomerang: boomerangInterface;
-	icpswapPool: icpswapPoolInterface;
+		if (await authClient.isAuthenticated()) {
+			const identity = authClient.getIdentity();
+			const agent = HttpAgent.createSync({
+				identity,
+				host: HOST
+			});
+			canisters.set(await fetchActors(agent));
+			user.set(new User(identity.getPrincipal()));
+		} else {
+			await authClient.login({
+				maxTimeToLive: AUTH_MAX_TIME_TO_LIVE,
+				allowPinAuthentication: true,
+				derivationOrigin: DAPP_DERIVATION_ORIGIN,
+				identityProvider: IDENTITY_PROVIDER,
+				onSuccess: async () => {
+					const identity = authClient.getIdentity();
```

Comment:
Here, can't you pass the identity due to the onSuccess? 

```
onSuccess: async (identity) => {
```

Reply:
No, the onSuccess function does not provide a `type Identity` parameter.